source/read-write-pref.txt

```diff
@@ -0,0 +1,228 @@
+- Collection
+
+This list also indicates the increasing order of precedence of the option settings. For
+example, if you set a read concern level for a transaction, it will override a read
+concern level set for the client.
+
+These options allow you to customize the causal consistency and availability of the data
+in your replica sets.
+
+Write Concern
+-------------
+
+The write concern specifies the level of acknowledgement requested from MongoDB for 
+write operations, such as an insert or update, before the operation successfully returns.
+Operations that do not specify an explicit write concern inherit the global default write
+concern settings.
+
+For more information, see :manual:`Write Concern </reference/write-concern/>` in the
+Server manual. For detailed API documentation, see the `Write Concern API documentation
+<https://www.mongodb.com/docs/ruby-driver/current/api/Mongo/Client.html#write_concern-instance_method>`__.
+
+The following table describes the ``write_concern`` parameters:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 25 50
+
+   * - Parameter
+     - Type
+     - Description
+  
+   * - ``w`` *(optional)*
+     - integer or string
+     - Requests acknowledgment that the write operation has propagated to a specified
+       number of ``mongod`` instances or to ``mongod`` instances that are labelled
+       specified tags.
+
+   * - ``wtimeoutMS`` *(optional)*
+     - integer
+     - Specifies a time limit to prevent write operations from blocking indefinitely.
+
+   * - ``journal`` *(optional)*
+     - boolean
+     - Requests acknowledgment that the write operation has been written to the 
+       on-disk journal.
+
+Example: Set the Write Concern for a Single Write Operation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This code uses custom ``write_concern`` settings while creating new a document:
+
+.. literalinclude:: /includes/usage-examples/read-write-pref.rb
+   :language: ruby
+   :dedent:
+   :start-after: start-write-concern
+   :end-before: end-write-concern
+
+Example: Retrieve and Apply an Existing Write Concern
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This code uses the ``new_write_concern`` method to construct a ``write_concern`` from
+the options of an existing database reference, ``myDB``. ``myDB`` can be 
+replaced with a reference to any entity that accepts a write concern option. 
+Then the new write concern is applied to a document, ``myDoc``:
+
+.. literalinclude:: /includes/usage-examples/read-write-pref.rb
+   :language: ruby
+   :dedent:
+   :start-after: start-write-concern-2
+   :end-before: end-write-concern-2
+
+Read Concern
+------------
+
+The read concern specifies the following behaviors:
+
+- Level of :manual:`causal consistency </core/causal-consistency-read-write-concerns/>`
+  across replica sets.
+
+- `Isolation guarantees <https://www.mongodb.com/docs/current/core/read-isolation-consistency-recency/>`__ 
+  maintained during a query.
```

Comment:
No periods since these aren't complete sentences
```suggestion
- Level of :manual:`causal consistency </core/causal-consistency-read-write-concerns/>`
  across replica sets

- `Isolation guarantees <https://www.mongodb.com/docs/current/core/read-isolation-consistency-recency/>`__ 
  maintained during a query
```